### PR TITLE
Update/ga4 backend destinations

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Removes the feature flag for GA4 back-end events.

Add the ability to send the events to more than one property via a filter.

### How to test the changes in this Pull Request:

1. Have your site connected with two different properties in GA
2. Make sure you are running Newspack Manager on https://github.com/Automattic/newspack-manager/pull/128
3. Add the options that will set the "local" GA property:

```
update_option( 'ga4_measurement_protocol_secret', 'xxx' );
update_option( 'ga4_measurement_id', 'G-XXX' );
```

5. Add the option that will set the property that is handled by Newspack Manager:

```
update_option( 'newspack_ga4_info', [ 'measurement_id'=>'G-ZZZ, 'measurement_protocol_secret'=>'zzz'  ] );
```

6. Visit your site and open the two GA Real time dashboards
7. Interact with your site and perform actions that will trigger back-end events, like submitting forms from a prompt
8. Check that the back end events, such as `np_reader_registered` and `np_prompt_interaction` with a `form_submission_received` as the `action` param.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/data-events/connectors/ga4/class-ga4.php [hunk 1]
```diff
@@ -412,14 +406,37 @@ private static function send_event( Event $event, $client_id, $timestamp, $user_
...
 
-		wp_remote_post(
-			$url,
-			[
-				'body' => wp_json_encode( $payload ),
-			]
-		);
+		$destinations = [
+			self::get_credentials(),
+		];
```
Comment (miguelpeixe) [inline includes/data-events/connectors/ga4/class-ga4.php:411]: Shouldn't this array start empty and only include the local credentials if they are filled?

**Review by dkoo — COMMENTED** [review]
Wanted to note that I ran through the testing steps and all is working as described. I'll let @miguelpeixe approve based on his change requests.

File: includes/data-events/connectors/ga4/class-ga4.php (hunk 1 above)
Comment (leogermani) [reply includes/data-events/connectors/ga4/class-ga4.php]: > Shouldn't this array start empty and only include the local credentials if they are filled?

This is basically what happens. `get_credentials` will return an array with empty values.

> Also, what do you think of calling it $properties?

Agree. Changed in https://github.com/Automattic/newspack-plugin/pull/2503/commits/7cdf13a892053973f80dc1868a98ad2e92733ec9

Also updated the README and changed the filter name on the manager plugin -> https://github.com/Automattic/newspack-manager/pull/128/commits/83d82123e5a491e7b9b540d0c39f24a3d1950ffd

**Review by miguelpeixe — APPROVED** [review]
Thanks for revising, it tests well!

With the current strategy of an array with an empty value, I'm always getting the `Error sending event - np_prompt_interaction - Error: Missing GA4 API credentials.` because it tries to fetch the API URL when GA4 wasn't yet configured. If this is expected, it's ok to merge.

**Comment by leogermani** [comment]
> With the current strategy of an array with an empty value, I'm always getting the Error sending event - np_prompt_interaction - Error: Missing GA4 API credentials. because it tries to fetch the API URL when GA4 wasn't yet configured. If this is expected, it's ok to merge.

Yes, that's expected for now. Notice that this only gets outputted to the error log with debugging mode enabled.

I'm considering all this temporary as we still need to figure out how we'll handle the "local" credentials

**Pull request merged**